### PR TITLE
Refactor review-submission logic

### PR DESCRIPTION
**Pull request by geffrak**
JIRA: VMR-2289

DESCRIPTION

- Move all the nuke related logic in a hook.
- Add a basic render hook for the tk-maya
- Add Shotgun Create as a Submitter

### CODE REVIEW

File: python/tk_multi_reviewsubmission/actions.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+import os
```
[the remark this answers is not in the document]
Comment (geffrak) [reply python/tk_multi_reviewsubmission/actions.py]: Removed.

File: app.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 import sgtk.templatekey
 import copy
 import os
+import re
```
[the remark this answers is not in the document]
Comment (geffrak) [reply app.py]: Removed

File: hooks/tk-maya/render_media.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+
+import sgtk
+import tempfile
```
[the remark this answers is not in the document]
Comment (geffrak) [reply hooks/tk-maya/render_media.py]: Removed.

File: app.py [hunk 4]
```diff
@@ -30,7 +30,24 @@ def init_app(self):
...
-        pass
+
+        app = self.import_module("tk_multi_reviewsubmission")
+
+        display_name = self.get_setting("display_name")
+
+        menu_caption = "%s..." % display_name
+        menu_options = {
+            "short_name": "send_for_review",
+            "description": "Send a version for review Using Shotgun Create",
```
Comment (jfboismenu) [inline app.py]: Seems a bit look for a menu item name. I'm sure our UX peeps will want to chime in. Now sure how to @ them here...

File: hooks/render_media.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        :param color_space:     Colorspace of the input frames
+
+        :returns:               Location of the rendered media
+        :rtype:                 str
+        """
+
+        pass
+
+    def _get_temp_media_path(self, name, version):
+        name = name or ""
```
Comment (jfboismenu) [inline hooks/render_media.py]: Missing doc string

File: hooks/render_media.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+
+import sgtk
+import tempfile
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class RenderMedia(HookBaseClass):
+    def render(
```
Comment (jfboismenu) [inline hooks/render_media.py]: This method looks like something that should not be here. There is no default implementation for render, right?

File: hooks/tk-maya/render_media.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+            "forceOverwrite": True,  # Overwrite existing playblast files which may have the the same name as the one specified with the "-f" flag
+            "format": "qt",  # The format of the output of this playblast.
+        }
+
+        if not output_path:
+            output_path = self._get_temp_media_path(name, version)
+
+        playblast_args[
+            "filename"
+        ] = output_path  # The filename to use for the output of this playblast.
```
Comment (jfboismenu) [inline hooks/tk-maya/render_media.py]: You code looks a bit weird because you've added an end of line comment. If you put it on the line above the code then it becomes two lines and is much easier to understand what is going on

File: hooks/tk-maya/render_media.py [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+        if width:
+            playblast_args[
+                "width"
+            ] = width  # Width of the final image. This value will be clamped if larger than the width of the active window.
```
Comment (jfboismenu) [inline hooks/tk-maya/render_media.py]: Same here

File: hooks/tk-maya/render_media.py [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+        if height:
+            playblast_args[
+                "height"
+            ] = height  # Height of the final image. This value will be clamped if larger than the width of the active window.
```
Comment (jfboismenu) [inline hooks/tk-maya/render_media.py]: Same here

File: python/tk_multi_reviewsubmission/__init__.py [hunk 10]
```diff
@@ -8,5 +8,42 @@
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
-from .renderer import Renderer
-from .submitter import Submitter
+from .actions import Actions
+
+__all__ = ["send_for_review", "render_and_submit_version"]
```
Comment (jfboismenu) [inline python/tk_multi_reviewsubmission/__init__.py]: Why is this needed?

File: python/tk_multi_reviewsubmission/actions.py [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+        :param sg_publishes:    A list of shotgun published file objects to link the publish against.
+        :param sg_task:         A Shotgun task object to link against. Can be None.
+        :param comment:         A description to add to the Version in Shotgun.
+        :param thumbnail_path:  The path to a thumbnail to use for the version when the movie isn't
+                                being uploaded to Shotgun (this is set in the config)
+        :param progress_cb:     A callback to report progress with.
+        :param color_space:     The colorspace of the rendered frames
+        """
+
+        if progress_cb:
```
Comment (jfboismenu) [inline python/tk_multi_reviewsubmission/actions.py]: Adding the following would simplify your code a bit. What do you think?
```
# Wrap the method so we don't have to worry about process_cb being None
progress_cb = lambda *args: progress_cb is None or process_cb(*args)
```

File: app.py (hunk 4 above)
Comment (geffrak) [reply app.py]: This is not the display name, it's the description. Default is `Send for review` and renamed to  `Playblast to Shotgun Create` in Maya.

File: hooks/render_media.py (hunk 6 above)
Comment (geffrak) [reply hooks/render_media.py]: Oups !
